### PR TITLE
MIGENG-103 Reporting API - Workload Inventory Report

### PR DESCRIPTION
**Pull request by mrizzi**
https://jira.coreos.com/browse/MIGENG-103

WIP because without tests

### CODE REVIEW

**Comment by mrizzi** [comment]
With commit 2ce0a62
* added tests (with and without `page` and `size` parameters) for the new `/report/{id}/workload-inventory` endpoint
* added the missing tests for `/report/{id}` and `/report/{id}/initial-saving-estimation` as well
* following https://github.com/project-xavier/xavier-integration/pull/16/files#diff-508a34a6f039810be3427e43274e4ed4R51, switched the `<bean>` components from using `beanType` attribute to `ref` in order to have the tests to work
* fixed the tests' method names `mainRouteBuilder*` to be `xmlRouteBuilder*` to get consistency with the naming convention

File: src/main/java/org/jboss/xavier/integrations/jpa/repository/WorkloadInventoryReportRepository.java [hunk 1]
```diff
@@ -0,0 +1,15 @@
+package org.jboss.xavier.integrations.jpa.repository;
+
+import org.jboss.xavier.analytics.pojo.output.InitialSavingsEstimationReportModel;
```
Comment (carlosthe19916) [inline src/main/java/org/jboss/xavier/integrations/jpa/repository/WorkloadInventoryReportRepository.java]: This import is unnecessary

File: src/main/java/org/jboss/xavier/integrations/jpa/repository/WorkloadInventoryReportRepository.java [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+import org.jboss.xavier.analytics.pojo.output.WorkloadInventoryReportModel;
+import org.jboss.xavier.integrations.jpa.projection.InitialSavingsEstimationReportSummary;
```
Comment (carlosthe19916) [inline src/main/java/org/jboss/xavier/integrations/jpa/repository/WorkloadInventoryReportRepository.java]: This import is unnecessary

File: src/test/java/org/jboss/xavier/integrations/route/XmlRoutes_RestReportTest.java [hunk 3]
```diff
@@ -80,7 +84,87 @@ public void mainRouteBuilder_RestReport_NotSummaryParamGiven_ShouldCallFindRepor
...
+    public void xmlRouteBuilder_RestReportId_IdParamGiven_ShouldCallFindReportSummaryById() throws Exception {
+        //Given
+        camelContext.setTracing(true);
+        camelContext.setAutoStartup(false);
+
+        //When
+        camelContext.start();
+        camelContext.startRoute("report-get-details");
+        Map<String, Long> variables = new HashMap<>();
+        Long one = new Long(1);
```
Comment (carlosthe19916) [inline src/test/java/org/jboss/xavier/integrations/route/XmlRoutes_RestReportTest.java]: I would suggest changing this to `Long one = 1L;` but it is Ok if you just ignore this. There are a couple of lines with the same definition in this class.

**Review by carlosthe19916 — COMMENTED** [review]
In general, this PR looks great to me. It would be great to merge this PR.

File: src/main/java/org/jboss/xavier/integrations/jpa/repository/WorkloadInventoryReportRepository.java (hunk 1 above)
Comment (mrizzi) [reply src/main/java/org/jboss/xavier/integrations/jpa/repository/WorkloadInventoryReportRepository.java]: thanks, i'll remove it

File: src/main/java/org/jboss/xavier/integrations/jpa/repository/WorkloadInventoryReportRepository.java (hunk 2 above)
Comment (mrizzi) [reply src/main/java/org/jboss/xavier/integrations/jpa/repository/WorkloadInventoryReportRepository.java]: thanks, i'll remove it

File: src/test/java/org/jboss/xavier/integrations/route/XmlRoutes_RestReportTest.java (hunk 3 above)
Comment (mrizzi) [reply src/test/java/org/jboss/xavier/integrations/route/XmlRoutes_RestReportTest.java]: I'll change this :+1:

**Review by carlosthe19916 — APPROVED** [review]
This PR looks great to me. I tested it using the UI and postman and it worked well!

**Pull request merged**